tests/unit/s2n_early_data_test.c

```diff
@@ -596,6 +615,45 @@ int main(int argc, char **argv)
 
             EXPECT_SUCCESS(s2n_connection_free(conn));
         }
+
+        /* Triggers callback to let application reject early data */
+        {
+            struct s2n_config *config = s2n_config_new();
+            EXPECT_NOT_NULL(config);
+
+            struct s2n_connection *conn = s2n_connection_new(S2N_SERVER);
+            EXPECT_NOT_NULL(conn);
+            EXPECT_SUCCESS(s2n_connection_set_config(conn, config));
+            EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(conn, "default_tls13"));
+            EXPECT_OK(s2n_append_test_chosen_psk_with_early_data(conn, nonzero_max_early_data, &s2n_tls13_aes_256_gcm_sha384));
+            EXPECT_SUCCESS(s2n_connection_set_early_data_expected(conn));
+            conn->secure.cipher_suite = &s2n_tls13_aes_256_gcm_sha384;
+            conn->actual_protocol_version = S2N_TLS13;
+
+            /* Without callback set, accepts early data */
+            conn->early_data_state = S2N_EARLY_DATA_REQUESTED;
+            EXPECT_OK(s2n_early_data_accept_or_reject(conn));
+            EXPECT_EQUAL(conn->early_data_state, S2N_EARLY_DATA_ACCEPTED);
+
+            uint8_t reject_early_data = false;
```

Comment:
NitPick: Variables with negative names are confusing. I'd prefer to see this named `accept_early_data` with its meaning reverse.